adagrams/game.py

```diff
@@ -1,11 +1,151 @@
+    letters_dict = transform_to_dictionary_without_replacement(letter_bank)
+        
+    for key, value in word_dict.items():
+        if key in letters_dict:
+            letters_dict[key] = letters_dict[key] - value
+        else:
+            return False
+
+    # transforming letter_dict_values into a list to check if value is < 0
+    letter_dict_values = letters_dict.values()
+
+    for value in letter_dict_values:
+        if value < 0:
+            return False
+
+    return True
+
 
 def score_word(word):
-    pass
+    """
+    input: word (string of characters/letters)
+    output: number of points integer
+    """
+
+    letter_point_value_dict = {
+        "A": 1,
+        "E": 1,
+        "I": 1,
+        "O": 1,
+        "U": 1,
+        "L": 1,
+        "N": 1,
+        "R": 1,
+        "S": 1,
+        "T": 1,
+        "D": 2,
+        "G": 2,
+        "B": 3,
+        "C": 3,
+        "M": 3,
+        "P": 3,
+        "F": 4,
+        "H": 4,
+        "V": 4,
+        "W": 4,
+        "Y": 4,
+        "K": 5,
+        "J": 8,
+        "X": 8,
+        "Q": 10,
+        "Z": 10
+    } 
+
+    user_score = 0
+    long_word_bonus = 8
+
+    if len(word) > 10 or len(word) == 0:
+        return False
+    
+    if word is not word.upper():
+        word = word.upper()
+
+    # updates user_score through every iteration of letter in word
+    for letter in word:
+        user_score = user_score + letter_point_value_dict[letter]
+            
+    # if checks if user will get long_word_bonus
+    if len(word) > 6:
+        user_score = user_score + long_word_bonus
+
+
+    return user_score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    """
+    input: word_list (list of strings)
+    output: word string, and score of word (tuple) 
+    """
+    word_scores_list_of_dicts = []
+    ten_letter_words = []
+
+    # creates list of dicts where each dict represents a word w/ two key-value pairs
+    for word in word_list:
+        word_scores_list_of_dicts.append({
+            "word": word,
+            "score": score_word(word)
+        })
+
+    # max only returns ONE dictionary, will use it to compare to other dicts later
+    # key (optional param for max) func where iterables are passed and comparison is performed based on its return value
+    best_word = max(word_scores_list_of_dicts, key=lambda word_dict: word_dict["score"])
+
+    for word_dict in word_scores_list_of_dicts:
+        # handles word with length of 10 letters
+        if len(word_dict["word"]) == 10:
+            ten_letter_words.append(word_dict)
+
+        # handles tie logic 
+        if word_dict["score"] == best_word["score"]:
+            if len(word_dict["word"]) < len(best_word["word"]):
+                best_word = word_dict
+
+    if len(ten_letter_words) > 0:
+        best_word = ten_letter_words[0]
+    
+    return tuple([best_word['word'], best_word['score']])
```

Comment:
Great job casting this into a tuple in the `return` statement here! 

One tiny nit: add a blank line at the end of your files!